FlowCrypt/src/main/java/com/flowcrypt/email/api/email/model/IncomingMessageInfo.kt

```diff
@@ -84,6 +91,140 @@ data class IncomingMessageInfo constructor(
     return hasSomePart(MsgBlock.Type.PLAIN_TEXT)
   }
 
+  fun toInitializationData(
+    context: Context,
+    @MessageType messageType: Int,
+    accountEmail: String,
+    aliases: List<String>
+  ): InitializationData {
+    val toAddresses = arrayListOf<String>()
+    val ccAddresses = arrayListOf<String>()
+    val bccAddresses = arrayListOf<String>()
+    var body: String? = null
+
+    val folderType = FoldersManager.getFolderType(localFolder)
+    when (messageType) {
+      MessageType.REPLY -> {
+        when (folderType) {
+          FoldersManager.FolderType.SENT,
+          FoldersManager.FolderType.OUTBOX -> {
+            toAddresses.addAll(getTo().map { it.address.lowercase() })
+          }
+
+          else -> {
+            toAddresses.addAll(getReplyToWithoutOwnerAddress().ifEmpty { getTo() }
+              .map { it.address.lowercase() })
+          }
+        }
+      }
+
+      MessageType.REPLY_ALL -> {
+        when (folderType) {
+          FoldersManager.FolderType.SENT, FoldersManager.FolderType.OUTBOX -> {
+            toAddresses.addAll(getTo().map { it.address.lowercase() })
+            ccAddresses.addAll(getCc().map { it.address.lowercase() })
+          }
+
+          else -> {
+            val toRecipients = getReplyToWithoutOwnerAddress().ifEmpty { getTo() }
+            toAddresses.addAll(toRecipients.map { it.address.lowercase() })
+
+            val ccSet = LinkedHashSet<InternetAddress>()
+
+            if (getTo().isNotEmpty()) {
+              ccSet.addAll(getTo().filter { !accountEmail.equals(it.address, ignoreCase = true) })
+
+              for (alias in aliases) {
+                val iterator = ccSet.iterator()
+                while (iterator.hasNext()) {
+                  if (iterator.next().address.equals(alias, ignoreCase = true)) {
+                    iterator.remove()
+                  }
+                }
+              }
+            }
```

Comment:
If I understand correctly, this is to remove aliasEmails from ccSet.
I think we can use below logic to filter out.

```kt
// couldn't use `aliases.contain` because it doesn't support `ignoreCase`
ccSet.addAll(getTo().filter { !accountEmail.equals(it.address, true) && !aliases.any { alias -> alias.equals(it.address, true) } })
```

instead of below

```kt
ccSet.addAll(getTo().filter { !accountEmail.equals(it.address, ignoreCase = true) })

for (alias in aliases) {
  val iterator = ccSet.iterator()
  while (iterator.hasNext()) {
    if (iterator.next().address.equals(alias, ignoreCase = true)) {
      iterator.remove()
    }
  }
}
```

Reply:
> Of course, it can be simplified. Please pay attention that firstly the whole java code was converted to Kotlin via Android Studio. There are a lot of things that look not optimized.

I hope all of these can be identified, as Ioan did, and then fixed. It will make the codebase a lot easier to follow. @DenBond7 please file an issue for you to go over the whole codebase and looks for similar opportunities to simplify. It will be a lot of work, but worthwhile.

@ioanmo226 please do highlight any other places like this that you notice, and we can file them all for fixing / improvements.

Reply:
https://github.com/FlowCrypt/flowcrypt-android/issues/2005